app/src/main/java/org/xwiki/android/sync/AppContext.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.android.sync
+
+import android.app.Application
+import android.util.Log
+import org.xwiki.android.sync.rest.BaseApiManager
+import org.xwiki.android.sync.utils.SharedPrefsUtils
+
+import java.util.AbstractMap
+import java.util.ArrayList
+
+/**
+ * Application class for authenticator
+ *
+ * @version $Id: c3a5996b1bce14d5c105a55f085115347c39c035 $
+ */
+class AppContext : Application() {
+
+    /**
+     * Set [.instance] to this object.
+     */
+    override fun onCreate() {
+        super.onCreate()
+        instance = this
+        Log.d(TAG, "on create")
+    }
+
+    companion object {
+
+        /**
+         * Entry pair Server address - Base Api Manager
+         */
+        private var baseApiManager: AbstractMap.SimpleEntry<String, BaseApiManager>? = null
```

Comment:
Is there some reason to keep it here instead of out of the class in one file?

Reply:
As the baseAPiManager was declared as static, I'd put it in the companion object.

Reply:
In case of external value and function, it also will be static. The difference will be just in the package of the fun/val. I think, it will be more clear in code if you will try to avoid usage of companions in most cases